pkg/config/expand.go

```diff
@@ -510,39 +411,29 @@ func validateModuleSettingReference(bp Blueprint, mod Module, r Reference) error
 		return nil
 	}
 	g := bp.ModuleGroupOrDie(mod.ID)
-	callingModuleGroupIndex := slices.IndexFunc(bp.DeploymentGroups, func(d DeploymentGroup) bool { return d.Name == g.Name })
 
-	targetModuleGroupIndex, err := modToGrp(bp.DeploymentGroups, r.Module)
+	tm, err := bp.Module(r.Module)
```

Comment:
I believe this block will become self-explanatory (no comment needed), if you change the func name to `GetModule` as advised elsewhere. Otherwise, this block needs a comment like:

```go
// error if reference target module not found
```

Reply:
Made it less cumbersome by ignoring error.